server/src/main/java/org/apache/iotdb/db/service/thrift/impl/InternalServiceImpl.java

```diff
@@ -365,22 +363,33 @@ public TSStatus invalidatePermissionCache(TInvalidatePermissionCacheReq req) thr
 
   @Override
   public TSStatus deleteRegion(TConsensusGroupId tconsensusGroupId) throws TException {
-    QueryId queryId = Coordinator.getInstance().createQueryId();
-    PlanNodeId planNodeId = queryId.genPlanNodeId();
-    DeleteRegionNode deleteRegionNode = new DeleteRegionNode(queryId.genPlanNodeId());
     ConsensusGroupId consensusGroupId =
         ConsensusGroupId.Factory.createFromTConsensusGroupId(tconsensusGroupId);
-    deleteRegionNode.setConsensusGroupId(consensusGroupId);
-    deleteRegionNode.setPlanNodeId(planNodeId);
     if (consensusGroupId instanceof DataRegionId) {
-      return DataRegionConsensusImpl.getInstance()
-          .write(consensusGroupId, deleteRegionNode)
-          .getStatus();
+      ConsensusGenericResponse response =
+          DataRegionConsensusImpl.getInstance().removeConsensusGroup(consensusGroupId);
+      if (!response.isSuccess()
+          && !(response.getException() instanceof PeerNotInConsensusGroupException)) {
+        return RpcUtils.getStatus(
+            TSStatusCode.INTERNAL_SERVER_ERROR, response.getException().getMessage());
```

Comment:
Not use INTERNAL_ERROR.  Maybe choose other detailed TSStatusCode, like DELETE_REGION_ERROR.

Reply:
Fixed